component/plugincontroller/component.go

```diff
@@ -0,0 +1,34 @@
+package plugincontroller
+
+import (
+	"github.com/docker/docker/component"
+	"github.com/docker/docker/plugin"
+	"golang.org/x/net/context"
+)
+
+const name = "plugincontroller"
+
+// Set sets the plugin controller component
+func Set(s *plugin.Manager) (cancel func(), err error) {
+	return component.Register(name, s)
+}
```

Comment:
This seems like a strange/unnecessary pattern.  Why not just register it directly from where it's created?

```go
registry.Register(name, pluginManager)
```

What's the purpose of the return values?

I guess `cancel()` is for removing a component. Why would that be necessary?